myna/database/nist_ambench_2022.py

```diff
@@ -0,0 +1,537 @@
+                                "Y(mm)": pl.Series([row["Y"]], dtype=pl.Float64),
+                                "Z(mm)": pl.Series([0], dtype=pl.Float64),
+                                "Pmod": pl.Series([0], dtype=pl.Float64),
+                                "tParam": pl.Series([0], dtype=pl.Float64),
+                            }
+                        )
+
+                    # If the next time is farther away than the SCAN_TIMESTEP, go to start point
+                    elif np.abs(row["dt_next"] - SCAN_TIMESTEP) > abs_tol:
+
+                        df_row = pl.DataFrame(
+                            {
+                                "Mode": pl.Series([1], dtype=pl.Int8),
+                                "X(mm)": pl.Series([row["X"]], dtype=pl.Float64),
+                                "Y(mm)": pl.Series([row["Y"]], dtype=pl.Float64),
+                                "Z(mm)": pl.Series([0], dtype=pl.Float64),
+                                "Pmod": pl.Series([0], dtype=pl.Float64),
+                                "tParam": pl.Series([row["dt_last"]], dtype=pl.Float64),
+                            }
+                        )
+
+                    # If the last time was longer ago than the SCAN_TIMESTEP, go raster to point
+                    elif np.abs(row["dt_last"] - SCAN_TIMESTEP) > abs_tol:
+
+                        df_row = pl.DataFrame(
+                            {
+                                "Mode": pl.Series([0], dtype=pl.Int8),
+                                "X(mm)": pl.Series([row["X"]], dtype=pl.Float64),
+                                "Y(mm)": pl.Series([row["Y"]], dtype=pl.Float64),
+                                "Z(mm)": pl.Series([0], dtype=pl.Float64),
+                                "Pmod": pl.Series([power], dtype=pl.Float64),
+                                "tParam": pl.Series(
+                                    [
+                                        row["dist_to_last"]
+                                        * 1e-3
+                                        / (row["Time(s)"] - time_last_row)
+                                    ],
+                                    dtype=pl.Float64,
+                                ),
+                            }
+                        )
+
+                    df_scan = df_scan.vstack(df_row)
+                    time_last_row = row["Time(s)"]
+
+                # Export to .txt file with tab-separated values
+                df_scan.write_csv(file_database, separator="\t")
+
+        return file_database
+
+    def get_plate_size(self):
+        """Load the (x,y) build plate size in meters"""
+        return np.array([0.1, 0.1])
+
+    def get_sync_image_size(self):
+        """Load the (x,y) image size in pixels"""
+        return np.array([2000, 2000])
+
+    def sync(self, component_type, step_types, output_class, files):
+        """Sync result files to the database using Peregrine-style directory
```

Comment:
I assume they don't have a specific way to save their data

Reply:
You can download the entire data archive as a zip file, so I that could be the default data structure. The scan path file is 99% of the data volume, so it's not unreasonable to have that be the expectation.

Reply:
Just realizing I probably misunderstood the question. They do have the templates for the challenge problem, but otherwise, there is no standard way that they process their data. It may be worthwhile to implement the challenge problem templates as an output, but I'm not sure the best way to do that and will probably leave it to a future PR.